consonance-arch/src/main/java/info/consonance/arch/jobGenerator/JobGenerator.java

```diff
@@ -55,16 +58,30 @@ public static void main(String[] argv) throws IOException {
     public JobGenerator(String[] argv) throws IOException {
         super();
 
-        this.iniDirSpec = super.parser.accepts("ini-dir", "schedule a batch of ini files from this directory").withOptionalArg()
-                .ofType(String.class);
-        this.workflowNameSpec = super.parser.accepts("workflow-name", "track the name of workflows").withOptionalArg().ofType(String.class)
-                .required();
-        this.workflowVersionSpec = super.parser.accepts("workflow-version", "track the version of workflows").withOptionalArg()
-                .ofType(String.class).required();
-        this.workflowPathSpec = super.parser.accepts("workflow-path", "Schedule workflows at this path on the container host")
+        ArgumentAcceptingOptionSpec<String> iniDirSpec = super.parser
+                .accepts("ini-dir", "schedule a batch of ini files from this directory").withRequiredArg().ofType(String.class);
+        ArgumentAcceptingOptionSpec<String> workflowNameSpec = super.parser.accepts("workflow-name", "track the name of workflows")
+                .withOptionalArg().ofType(String.class).required();
+        ArgumentAcceptingOptionSpec<String> workflowVersionSpec = super.parser.accepts("workflow-version", "track the version of workflows")
                 .withOptionalArg().ofType(String.class).required();
-        this.totalJobSpec = super.parser.accepts("total-jobs", "Schedule a specific number of test workflows")
-                .requiredUnless(iniDirSpec, this.endlessSpec).withRequiredArg().ofType(Integer.class).defaultsTo(Integer.MAX_VALUE);
+        ArgumentAcceptingOptionSpec<String> workflowPathSpec = super.parser
+                .accepts("workflow-path", "Schedule workflows at this path on the container host").withRequiredArg().ofType(String.class)
+                .required();
+        ArgumentAcceptingOptionSpec<Integer> totalJobSpec = super.parser
+                .accepts("total-jobs", "Schedule a specific number of test workflows").requiredUnless(iniDirSpec, this.endlessSpec)
+                .withRequiredArg().ofType(Integer.class).defaultsTo(Integer.MAX_VALUE);
+
+        ArgumentAcceptingOptionSpec<String> userSpec = super.parser
+                .accepts("user", "Designate a user for the jobs submitted in this batch").withRequiredArg().ofType(String.class)
+                .defaultsTo("Player");
+        ArgumentAcceptingOptionSpec<String> flavourSpec = super.parser
+                .accepts("flavour", "Designate a specific flavour for the jobs submitted in this batch").withRequiredArg().ofType(String.class)
+                .required();
+
+        ArgumentAcceptingOptionSpec<KeyValuePair> extraFilesSpec = super.parser
+                .accepts("extra-files", "Submit extra files that should exist on the worker when executing. This should be key values ").withRequiredArg().ofType(KeyValuePair.class)
```

Comment:
Pretty much anything that the container author wants to be mounted but doesn't want to share in the container. Probably creds, but its up to them.